x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -309,13 +317,20 @@ public boolean test(PersistentTasksCustomMetadata.PersistentTask<?> persistentTa
                     node = persistentTask.getExecutorNode();
                     return true;
                 case CLOSING:
-                    exception = ExceptionsHelper.conflictStatusException("The job has been " + JobState.CLOSED + " while waiting to be "
-                            + JobState.OPENED);
+                    exception = ExceptionsHelper.conflictStatusException(
+                        "The job has been {} while waiting to be {}",
+                        JobState.CLOSED,
+                        JobState.OPENED);
                     return true;
                 case FAILED:
                 default:
-                    exception = ExceptionsHelper.serverError("Unexpected job state [" + jobState
-                            + "] while waiting for job to be " + JobState.OPENED);
+                    // Default http status is SERVER ERROR
+                    exception = ExceptionsHelper.serverError(
+                        "Unexpected job state [{}] with reason [{}] while waiting for job to be {}",
```

Comment:
Can you change the message to include reason only if it is known please. 

"Unexpected job state [XX] with reason [__UNKNOWN__]... " is a message that can only frustrate